internal/workspace/git.go

```diff
@@ -4,10 +4,22 @@ import (
 	"context"
 	"fmt"
 	"net/http"
+	"strings"
 
 	"github.com/google/go-github/v72/github"
 )
 
+// InsufficientPermissionsError indicates that an operation failed due to insufficient GitHub token permissions.
+// This error is intended to be converted to a ToolInputError by the bot layer, prompting the bot to report a limitation.
+type InsufficientPermissionsError struct {
+	Operation string
+	Reason    string
+}
+
+func (ipe InsufficientPermissionsError) Error() string {
+	return fmt.Sprintf("insufficient permissions for %s: %s", ipe.Operation, ipe.Reason)
```

Comment:
Now the grammar is wrong here - "insufficient permissions for modify GitHub workflow files"